frontend/packages/helm-plugin/src/actions/providers.ts

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { getHelmDeleteAction, getHelmRollbackAction, getHelmUpgradeAction } from './creators';
+import { HelmActionsScope } from './types';
+
+export const useHelmActionProvider = (scope: HelmActionsScope) => {
+  const { t } = useTranslation();
+  const actions = React.useMemo(
+    () => [
+      getHelmUpgradeAction(scope, t),
+      getHelmRollbackAction(scope, t),
+      getHelmDeleteAction(scope, t),
+    ],
+    [scope, t],
```

Comment:
Can you be sure that `scope` will not be mutated by the caller?

Reply:
That would be the assumption here since the provider and creators are all owned by the same plugin contributing these actions.